extra-dev/packages/coq-dpdgraph/coq-dpdgraph.dev/opam

```diff
@@ -0,0 +1,42 @@
+opam-version: "2.0"
+maintainer: "palmskog@gmail.com"
+
+homepage: "https://github.com/coq-community/coq-dpdgraph"
+dev-repo: "git+https://github.com/coq-community/coq-dpdgraph.git"
+bug-reports: "https://github.com/coq-community/coq-dpdgraph/issues"
+license: "LGPL-2.1-only"
+
+synopsis: "Compute dependencies between Coq objects (definitions, theorems) and produce graphs"
+description: """
+Coq plugin that extracts the dependencies between Coq objects,
+and produces files with dependency information. Includes tools
+to visualize dependency graphs and find unused definitions."""
+
+build: [
+  ["autoreconf" "-i" "-s"]
+  ["./configure"]
+  [make "-j%{jobs}%" "WARN_ERR="]
+]
+install: [make "install" "BINDIR=%{bin}%"]
+depends: [
+  "ocaml" {>= "4.05.0"}
+  "coq" {>= "8.15" & < "8.16~" | = "dev"}
+  "ocamlgraph"
+  "conf-automake"
+]
```

Comment:
```suggestion
build: [
  ["autoconf"] {dev}
  ["./configure"]
  [make "-j%{jobs}%" "WARN_ERR="]
]
run-test: [make "test-suite"]
install: [make "install" "BINDIR=%{bin}%"]
depends: [
  "ocaml" {>= "4.05.0"}
  "coq" {= "dev"}
  "conf-autoconf" {build & dev}
  "ocamlgraph" 
]
```
Here is what we actually use in CI.

Reply:
I don't like it to depend dev packages on dev coq. Why shouldn't one be allowed to use a dev package on a release coq - as long as it works?

Reply:
Let's be clear here about expectations. I want the `dev` packages in `extra-dev` to conform *as much as possible* to the opam packages in the repositories, for maintenance reasons. Sometimes small divergences are unavoidable, but you are here proposing having different dependencies (and build systems). coq-dpdgraph's master is a plugin that is expected to be compatible with current stable Coq only by accident.

Reply:
As I explained this cannot be avoided. The usual opam package uses a tar ball which requires manual steps to create it (as discussed in the dpdgraph issue). A dev opam package must do these otherwise manual steps as well. This requires the dependencies used to do these steps.

Also as discussed in the dpdgraph issue I don't like opam packages which use manually processed tar balls. I would prefer release tar balls which contain all build steps from the git tag to the executable.

But as is it is technically impossible to make the dev package build and dependecy wise equivalent to the release package.

Reply:
The discussion: (https://github.com/coq-community/coq-dpdgraph/issues/95)

Reply:
Sure, I remember that discussion. But there are several issues here with the `coq-dpdgraph.dev` package you are proposing, which were not covered in that discussion:
1. It doesn't use the same build system as in the repository
2. It doesn't use the same dependencies as in the repository (plugins are under heavy churn and will break compatibility with stable Coq versions quickly)

Recall that we use automation in the form of the templates to *generate* the opam package in the GitHub repository. Significantly diverging opam packages between `extra-dev` and GitHub repos will make long term maintenance really difficult, and we even have an issue about it: https://github.com/coq/opam-coq-archive/issues/995

It's possible that the autoconf dependency will be lifted for coq-dpdgraph in the future, but then we should still aim to synchronize packages between `extra-dev` and the GitHub repo.

Reply:
OK let me double check this. I thought I just did take the latest release package and did the minimal required changes - maybe I messed something up in the process.

Reply:
If you need a `dev` package for a plugin that works with a stable version of Coq, this should be handled by adding a `8.XY.dev` package that uses a plugin repo branch, as here: https://github.com/coq/opam-coq-archive/tree/master/extra-dev/packages/coq-ltac2/coq-ltac2.8.10.dev

Reply:
I reviewed my dev package against the latest release package for dpdgraph. The differences are:

- added `["autoreconf" "-i" "-s"]` - either this or autoconf is absolutely required - this step is done manually in the release packages, so this is just a virtual difference. It shouldn't really matter if this is done by a human before generating the tar ball or automatically by opam - this step is always done. And if the source is git, doing it automatically is the only option.
- added the corresponding dependency on `conf-automake` (could be `conf-autoconf`)
- removed the date tag
- use git as source

So I really don't know what e.g. "It doesn't use the same build system as in the repository" refers to.

Reply:
Do you agree with the following?

- `automake` and `autoconf` are different tools (and live in different packages in OS distributions such as Ubuntu)
- project repository CI and the `extra-dev` package should run the exact same build commands

If you agree, then I think you should apply my change suggestion for the `coq-dpdgraph.dev` package. Otherwise, I guess we are at an impasse, and I will leave to someone else to decide on merging.

Reply:
How do you know which tool @ybertot calls manually? I don't think the release package documents this in any way, so what you say is just based on assumptions.

But if you have a problem with automake (as I said in my experience it is more robust than autoconf - but as @silene pointed out likely not in this case) we can switch to autoconf.

The main thing I didn't like in your change was that it restricts the package to coq dev.

Reply:
I did the last two releases for coq-dpdgraph before Yves did the one for 8.15, and I set up the CI and opam package in the repo (inspired by Flocq). But again my main concern is concistency between repos and `extra-dev`.

As for the Coq version, this is a long standing policy for plugins - that they should only be considered compatible with a single major version of Coq (`dev` or 8.XY). coq-dpdgraph's branch for Coq master probably only has a few weeks more of compatibility with Coq 8.15.

Reply:
> But again my main concern is consistency between repos and `extra-dev`.

What are we discussing here? autoconf vs autoreconf or somethign else? If it is autoconf vs autoreconf as I said a few times I have no issue to change it. If it is something else I have not the slightest idea what inconsistency you are talking about. The only difference is that in dev I call autoreconf/autoconf and that this is done manually in the release packages. If you want to have consistency in this point the only technically possible way is to change the release packages - the dev packages must automate this step.

> As for the Coq version, this is a long standing policy for plugins - that they should only be considered compatible with a single major version of Coq (dev or 8.XY)

OK, that makes sense. I was not aware that there is a different policy for plugins and other developments.

Reply:
Since we agreed on all points, I committed your change. But I would still like to understand what "inconsistency" you are talking about.